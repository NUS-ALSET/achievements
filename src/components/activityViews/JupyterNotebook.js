/**
 * @file JupyterNotebook container module
 * @author Theodor Shaytanov <theodor.shaytanov@gmail.com>
 * @created 09.06.18
 */
/* eslint-disable react/display-name */

import React from "react";
import PropTypes from "prop-types";
import ReactLoadable from "react-loadable";

import Button from "@material-ui/core/Button";
import CircularProgress from "@material-ui/core/CircularProgress";
import Collapse from "@material-ui/core/Collapse";
import IconButton from "@material-ui/core/IconButton";
import InputAdornment from "@material-ui/core/InputAdornment";
import LinearProgress from "@material-ui/core/LinearProgress";
import Paper from "@material-ui/core/Paper";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";

import ExpandLessIcon from "@material-ui/icons/ExpandLess";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import OpenInNewIcon from "@material-ui/icons/OpenInNew";
import RefreshIcon from "@material-ui/icons/Refresh";

import { APP_SETTING } from "../../achievementsApp/config";

const AceEditor = ReactLoadable({
  loader: () => import("../AceEditor"),
  loading: () => <LinearProgress />
});

const NotebookPreview = ReactLoadable({
  loader: () => import("@nteract/notebook-preview"),
  loading: () => <LinearProgress />
});

class JupyterNotebook extends React.PureComponent {
  static propTypes = {
    action: PropTypes.func,
    defaultValue: PropTypes.string,
    persistent: PropTypes.bool,
    richEditor: PropTypes.bool,
    solution: PropTypes.any,
    title: PropTypes.any.isRequired,
    url: PropTypes.string,
    readOnly : PropTypes.bool
  };

  state = {
    collapsed: false,
    solution: ""
  };

  onChange = e =>
    this.setState({ solution: (e && e.target && e.target.value) || e });

  onSwitchCollapse = () => {
    this.setState({
      collapsed: !this.state.collapsed
    });
  };

  render() {
    const {
      action,
      defaultValue,
      persistent,
      richEditor,
      solution,
      title,
      url,
      readOnly
    } = this.props;
    return (
      <Paper style={{ margin: "24px 2px" }}>
        <Typography
          style={{
            position: "relative"
          }}
          variant="headline"
        >
          <span>{title}</span>
          {url && (
            <IconButton
              style={{
                position: "absolute",
                right: 48
              }}
            >
              <a
                href={url}
                style={{ position: "absolute", top: 10 }}
                target="_blank"
              >
                <OpenInNewIcon />
              </a>
            </IconButton>
          )}
          {persistent ? (
            action &&
            this.state.solution &&
            richEditor &&
            (APP_SETTING.isSuggesting ? (
              <IconButton
                onClick={() => action(this.state.solution)}
                style={{
                  position: "absolute",
                  right: 0
                }}
              >
                <RefreshIcon />
              </IconButton>
            ) : (
              <Button
                color="primary"
                onClick={() => action(this.state.solution)}
                style={{
                  position: "absolute",
                  top: 4,
                  right: 4
                }}
              >
                Run
              </Button>
            ))
          ) : (
            <IconButton
              onClick={this.onSwitchCollapse}
              style={{
                position: "absolute",
                right: 0
              }}
            >
              {this.state.collapsed ? <ExpandLessIcon /> : <ExpandMoreIcon />}
            </IconButton>
          )}
        </Typography>
        {solution !== null &&
          action &&
          (richEditor ? (
            <AceEditor
              editorProps={{ $blockScrolling: true }}
              maxLines={20}
              minLines={10}
              mode="python"
              onChange={this.onChange}
              onLoad={editor => editor.focus()}
              theme="github"
              value={this.state.solution || defaultValue || ""}
              readOnly={readOnly}
            />
          ) : (
            <TextField
              defaultValue={this.state.solution || defaultValue || ""}
              fullWidth
              InputLabelProps={{
                style: {
                  top: 24,
                  left: 24
                }
              }}
              InputProps={{
                endAdornment: (
                  <InputAdornment position="end">
                    {APP_SETTING.isSuggesting ? (
                      <IconButton onClick={() => action(this.state.solution)}>
                        <RefreshIcon />
                      </IconButton>
                    ) : (
                      <Button
                        color="primary"
                        onClick={() => action(this.state.solution)}
                        style={{
                          marginBottom: 4
                        }}
                      >
                        Run
                      </Button>
                    )}
                  </InputAdornment>
                )
              }}
              label="Enter the url to your public solution on Colab"
              onChange={this.onChange}
              style={{ padding: 24, position: "relative" }}
              disabled={readOnly}
            />
          ))}
        {solution &&
          solution.json && (
            <Collapse collapsedHeight="10px" in={!this.state.collapsed}>
              <div
                style={{
                  textAlign: "left"
                }}
              >
                <NotebookPreview notebook={solution.json} />
              </div>
            </Collapse>
          )}
        {solution && solution.loading && <CircularProgress />}
      </Paper>
    );
  }
}

export default JupyterNotebook;
