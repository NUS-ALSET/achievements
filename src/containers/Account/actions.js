export const ACCOUNT_OPEN = "ACCOUNT_OPEN";
export const accountOpen = accountId => ({
  type: ACCOUNT_OPEN,
  accountId
});

export const ACCOUNT_FETCH_PATHS = "ACCOUNT_FETCH_PATHS";
export const accountFetchPaths = (accountId, paths) => ({
  type: ACCOUNT_FETCH_PATHS,
  accountId,
  paths
});

export const EXTERNAL_PROFILE_DIALOG_SHOW = "EXTERNAL_PROFILE_DIALOG_SHOW";

export const ACCOUNT_CHANGE_ADMIN_STATUS = "ACCOUNT_CHANGE_ADMIN_STATUS";
export const accountChangeAdminStatus = adminStatus => ({
  type: ACCOUNT_CHANGE_ADMIN_STATUS,
  adminStatus
});

/**
 * Creates action for external profile dialog show
 *
 * @param {Object} externalProfile
 * @param {String} externalProfile.name
 * @returns {{type: string, externalProfile: *}} action
 */
export const externalProfileDialogShow = externalProfile => ({
  type: EXTERNAL_PROFILE_DIALOG_SHOW,
  externalProfile
});

export const EXTERNAL_PROFILE_DIALOG_HIDE = "EXTERNAL_PROFILE_DIALOG_HIDE";
export const externalProfileDialogHide = () => ({
  type: EXTERNAL_PROFILE_DIALOG_HIDE
});

export const EXTERNAL_PROFILE_UPDATE_REQUEST =
  "EXTERNAL_PROFILE_UPDATE_REQUEST";
export const externalProfileUpdateRequest = (
  externalProfileId,
  externalProfileType,
  customUID
) => ({
  type: EXTERNAL_PROFILE_UPDATE_REQUEST,
  externalProfileId,
  externalProfileType,
  customUID
});

export const PROFILE_UPDATE_DATA_REQUEST = "PROFILE_UPDATE_DATA_REQUEST";
export const profileUpdateDataRequest = (field, data) => ({
  type: PROFILE_UPDATE_DATA_REQUEST,
  field,
  data
});

export const PROFILE_UPDATE_DATA_SUCCESS = "PROFILE_UPDATE_DATA_SUCCESS";
export const profileUpdateDataSuccess = (field, data, customUID) => ({
  type: PROFILE_UPDATE_DATA_SUCCESS,
  field,
  data,
  customUID
});

export const PROFILE_UPDATE_DATA_FAIL = "PROFILE_UPDATE_DATA_FAIL";
export const profileUpdateDataFail = (field, data, reason) => ({
  type: PROFILE_UPDATE_DATA_FAIL,
  field,
  data,
  reason
});

export const EXTERNAL_PROFILE_UPDATE_SUCCESS =
  "EXTERNAL_PROFILE_UPDATE_SUCCESS";
export const externalProfileUpdateSuccess = (
  externalProfileId,
  externalProfileType
) => ({
  type: EXTERNAL_PROFILE_UPDATE_SUCCESS,
  externalProfileId,
  externalProfileType
});

export const EXTERNAL_PROFILE_UPDATE_FAIL = "EXTERNAL_PROFILE_UPDATE_FAIL";
export const externalProfileUpdateFail = (
  externalProfileId,
  externalProfileType,
  reason
) => ({
  type: EXTERNAL_PROFILE_UPDATE_FAIL,
  externalProfileId,
  externalProfileType,
  reason
});

export const EXTERNAL_PROFILE_REFRESH_REQUEST =
  "EXTERNAL_PROFILE_REFRESH_REQUEST";
export const externalProfileRefreshRequest = (
  externalProfileId,
  externalProfileType,
  customUID=null
) => ({
  type: EXTERNAL_PROFILE_REFRESH_REQUEST,
  externalProfileId,
  externalProfileType,
  customUID
});

export const EXTERNAL_PROFILE_REFRESH_SUCCESS =
  "EXTERNAL_PROFILE_REFRESH_SUCCESS";
export const externalProfileRefreshSuccess = (
  externalProfileId,
  externalProfileType
) => ({
  type: EXTERNAL_PROFILE_REFRESH_SUCCESS,
  externalProfileId,
  externalProfileType
});

export const EXTERNAL_PROFILE_REFRESH_FAIL = "EXTERNAL_PROFILE_REFRESH_FAIL";
export const externalProfileRefreshFail = (
  externalProfileId,
  externalProfileType,
  reason
) => ({
  type: EXTERNAL_PROFILE_REFRESH_FAIL,
  externalProfileId,
  externalProfileType,
  reason
});

export const EXTERNAL_PROFILE_REMOVE_DIALOG_SHOW =
  "EXTERNAL_PROFILE_REMOVE_DIALOG_SHOW";
export const externalProfileRemoveDialogShow = (
  externalProfileId,
  externalProfileType
) => ({
  type: EXTERNAL_PROFILE_REMOVE_DIALOG_SHOW,
  externalProfileId,
  externalProfileType
});

export const EXTERNAL_PROFILE_REMOVE_DIALOG_HIDE =
  "EXTERNAL_PROFILE_REMOVE_DIALOG_HIDE";
export const externalProfileRemoveDialogHide = () => ({
  type: EXTERNAL_PROFILE_REMOVE_DIALOG_HIDE
});

export const EXTERNAL_PROFILE_REMOVE_REQUEST =
  "EXTERNAL_PROFILE_REMOVE_REQUEST";
export const externalProfileRemoveRequest = externalProfileType => ({
  type: EXTERNAL_PROFILE_REMOVE_REQUEST,
  externalProfileType
});

export const EXTERNAL_PROFILE_REMOVE_SUCCESS =
  "EXTERNAL_PROFILE_REMOVE_SUCCESS";
export const externalProfileRemoveSuccess = externalProfileType => ({
  type: EXTERNAL_PROFILE_REMOVE_SUCCESS,
  externalProfileType
});

export const EXTERNAL_PROFILE_REMOVE_FAIL = "EXTERNAL_PROFILE_REMOVE_FAIL";
export const externalProfileRemoveFail = externalProfileType => ({
  type: EXTERNAL_PROFILE_REMOVE_FAIL,
  externalProfileType
});

export const DISPLAY_NAME_EDIT_TOGGLE = "DISPLAY_NAME_EDIT_TOGGLE";
export const displayNameEditToggle = status => ({
  type: DISPLAY_NAME_EDIT_TOGGLE,
  status
});

export const DISPLAY_NAME_UPDATE_REQUEST = "DISPLAY_NAME_UPDATE_REQUEST";
export const displayNameUpdateRequest = name => ({
  type: DISPLAY_NAME_UPDATE_REQUEST,
  name
});

export const DISPLAY_NAME_UPDATE_SUCCESS = "DISPLAY_NAME_UPDATE_SUCCESS";
export const displayNameUpdateSuccess = () => ({
  type: DISPLAY_NAME_UPDATE_SUCCESS
});

export const DISPLAY_NAME_UPDATE_FAIL = "DISPLAY_NAME_UPDATE_FAIL";
export const displayNameUpdateFail = reason => ({
  type: DISPLAY_NAME_UPDATE_FAIL,
  reason
});

export const FETCH_USER_DATA = "FETCH_USER_DATA";
export const fetchUserData = () => ({
  type: FETCH_USER_DATA
})

export const FETCH_USER_DATA_SUCCESS = "FETCH_USER_DATA_SUCCESS";
export const fetchUserDataSuccess = data => ({
  type: FETCH_USER_DATA_SUCCESS,
  data
})

export const FETCH_USER_DATA_FAIL = "FETCH_USER_DATA_FAIL";
export const fetchUserDataFail = reason => ({
  type: FETCH_USER_DATA_FAIL,
  reason
});

export const INSPECT_PATH_AS_USER = "INSPECT_PATH_AS_USER";
export const inspectPathAsUser = (pathId, userId) => ({
  type: INSPECT_PATH_AS_USER,
  pathId,
  userId
});
