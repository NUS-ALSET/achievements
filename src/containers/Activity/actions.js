export const PROBLEM_INIT_REQUEST = "PROBLEM_INIT_REQUEST";
export const problemInitRequest = (
  pathId,
  problemId,
  solution,
  readOnly = false
) => ({
  type: PROBLEM_INIT_REQUEST,
  problemId,
  pathId,
  solution,
  readOnly
});

export const PROBLEM_INIT_SUCCESS = "PROBLEM_INIT_SUCCESS";
export const problemInitSuccess = (
  pathId,
  problemId,
  payload,
  readOnly = false
) => ({
  type: PROBLEM_INIT_SUCCESS,
  problemId,
  pathId,
  payload,
  readOnly
});

export const PROBLEM_INIT_FAIL = "PROBLEM_INIT_FAIL";
export const problemInitFail = (pathId, problemId, reason) => ({
  type: PROBLEM_INIT_FAIL,
  problemId,
  pathId,
  reason
});

export const PROBLEM_FINALIZE = "PROBLEM_FINALIZE";
export const problemFinalize = () => ({
  type: PROBLEM_FINALIZE
});

export const PROBLEM_SOLVE_UPDATE = "PROBLEM_SOLVE_UPDATE";
export const problemSolveUpdate = (pathId, problemId, fileId, openTime = null) => {
  return ({
  type: PROBLEM_SOLVE_UPDATE,
  pathId,
  problemId,
  fileId,
  openTime
});}

export const PROBLEM_SOLVE_SUCCESS = "PROBLEM_SOLVE_SUCCESS";
export const problemSolveSuccess = (problemId, solutionKey) => ({
  type: PROBLEM_SOLVE_SUCCESS,
  problemId,
  solutionKey
});

export const PROBLEM_SOLUTION_REFRESH_REQUEST =
  "PROBLEM_SOLUTION_REFRESH_REQUEST";
export const problemSolutionRefreshRequest = (problemId, fileId, openTime) => ({
  type: PROBLEM_SOLUTION_REFRESH_REQUEST,
  problemId,
  fileId,
  openTime
});

export const PROBLEM_SOLUTION_REFRESH_SUCCESS =
  "PROBLEM_SOLUTION_REFRESH_SUCCESS";
export const problemSolutionRefreshSuccess = (problemId, payload) => ({
  type: PROBLEM_SOLUTION_REFRESH_SUCCESS,
  problemId,
  payload
});

export const PROBLEM_SOLUTION_EXECUTION_STATUS =
  "PROBLEM_SOLUTION_EXECUTION_STATUS";
export const problemSolutionExecutionStatus = payload => ({
  type: PROBLEM_SOLUTION_EXECUTION_STATUS,
  payload
});

export const PROBLEM_SOLUTION_CALCULATED_WRONG =
  "PROBLEM_SOLUTION_CALCULATED_WRONG";
export const problemSolutionCalculatedWrong = () => ({
  type: PROBLEM_SOLUTION_CALCULATED_WRONG
});

export const PROBLEM_SOLUTION_PROVIDED_SUCCESS =
  "PROBLEM_SOLUTION_PROVIDED_SUCCESS";
export const problemSolutionProvidedSuccess = (problemId, payload) => ({
  type: PROBLEM_SOLUTION_PROVIDED_SUCCESS,
  problemId,
  payload
});

export const PROBLEM_CHECK_SOLUTION_REQUEST = "PROBLEM_CHECK_SOLUTION_REQUEST";
export const problemCheckSolutionRequest = (problemId, fileId, solution) => ({
  type: PROBLEM_CHECK_SOLUTION_REQUEST,
  problemId,
  fileId,
  solution
});

export const PROBLEM_CHECK_SOLUTION_SUCCESS = "PROBLEM_CHECK_SOLUTION_SUCCESS";
export const problemCheckSolutionSuccess = (problemId, fileId, solution) => ({
  type: PROBLEM_CHECK_SOLUTION_SUCCESS,
  problemId,
  fileId,
  solution
});

export const PROBLEM_CHECK_SOLUTION_FAIL = "PROBLEM_CHECK_SOLUTION_FAIL";
export const problemCheckSolutionFail = (
  problemId,
  fileId,
  solution,
  reason
) => ({
  type: PROBLEM_CHECK_SOLUTION_FAIL,
  problemId,
  fileId,
  solution,
  reason
});

export const PROBLEM_SOLUTION_REFRESH_FAIL = "PROBLEM_SOLUTION_REFRESH_FAIL";
export const problemSolutionRefreshFail = (problemId, reason) => ({
  type: PROBLEM_SOLUTION_REFRESH_FAIL,
  problemId,
  reason
});

export const PROBLEM_SOLUTION_SUBMIT_REQUEST =
  "PROBLEM_SOLUTION_SUBMIT_REQUEST";
export const problemSolutionSubmitRequest = (pathId, problemId, payload) => ({
  type: PROBLEM_SOLUTION_SUBMIT_REQUEST,
  pathId,
  problemId,
  payload
});

export const PROBLEM_SOLUTION_SUBMIT_SUCCESS =
  "PROBLEM_SOLUTION_SUBMIT_SUCCESS";
export const problemSolutionSubmitSuccess = (pathId, problemId, payload) => ({
  type: PROBLEM_SOLUTION_SUBMIT_SUCCESS,
  pathId,
  problemId,
  payload
});

export const PROBLEM_SOLUTION_SUBMIT_FAIL = "PROBLEM_SOLUTION_SUBMIT_FAIL";
export const problemSolutionSubmitFail = (
  pathId,
  problemId,
  payload,
  reason
) => ({
  type: PROBLEM_SOLUTION_SUBMIT_FAIL,
  pathId,
  problemId,
  payload,
  reason
});

export const PROBLEM_SOLUTION_ATTEMPT_REQUEST =
  "PROBLEM_SOLUTION_ATTEMPT_REQUEST";
export const problemSolutionAttemptRequest = (activityKey, pathKey, activityType, completed, open, time) => ({
  type: PROBLEM_SOLUTION_ATTEMPT_REQUEST,
  payload: {
    activityKey,
    pathKey,
    activityType,
    completed,
    open,
    time
  }
});

export const SET_PROBLEM_OPEN_TIME = 'SET_PROBLEM_OPEN_TIME';
export const setProblemOpenTime = (problemId,openTime) => ({
  type: SET_PROBLEM_OPEN_TIME,
  problemId,
  openTime
})