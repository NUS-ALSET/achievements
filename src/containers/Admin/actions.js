export const ADMIN_UPDATE_CONFIG_REQUEST = "ADMIN_UPDATE_CONFIG_REQUEST";
export const adminUpdateConfigRequest = config => ({
  type: ADMIN_UPDATE_CONFIG_REQUEST,
  config
});

export const ADMIN_UPDATE_CONFIG_SUCCESS = "ADMIN_UPDATE_CONFIG_SUCCESS";
export const adminUpdateConfigSuccess = config => ({
  type: ADMIN_UPDATE_CONFIG_SUCCESS,
  config
});

export const ADMIN_UPDATE_CONFIG_FAIL = "ADMIN_UPDATE_CONFIG_FAIL";
export const adminUpdateConfigFail = (config, reason) => ({
  type: ADMIN_UPDATE_CONFIG_FAIL,
  config,
  reason
});

export const ADMIN_CUSTOM_AUTH_REQUEST = "ADMIN_CUSTOM_AUTH_REQUEST";
export const adminCustomAuthRequest = uid => ({
  type: ADMIN_CUSTOM_AUTH_REQUEST,
  uid
});

export const CREATE_NEW_SERVICE = "CREATE_NEW_SERVICE";
export const createNewService = data => ({
  type: CREATE_NEW_SERVICE,
  data
});

export const CREATE_NEW_SERVICE_SUCCESS = "CREATE_NEW_SERVICE_SUCCESS";
export const createNewServiceSuccess = () => ({
  type: CREATE_NEW_SERVICE_SUCCESS
});

export const CREATE_NEW_SERVICE_FALIURE = "CREATE_NEW_SERVICE_FALIURE";
export const createNewServiceFaliure = res => ({
  type: CREATE_NEW_SERVICE_FALIURE,
  res
});

export const FETCH_SERVICE_DETAILS = "FETCH_SERVICE_DETAILS";
export const fetchServiceDetails = id => ({
  type: FETCH_SERVICE_DETAILS,
  id
});

export const FETCH_SERVICE_DETAILS_SUCCESS = "FETCH_SERVICE_DETAILS_SUCCESS";
export const fetchServiceDetailsSuccess = service => ({
  type: FETCH_SERVICE_DETAILS_SUCCESS,
  service
});

export const FETCH_SERVICE_DETAILS_FALIURE = "FETCH_SERVICE_DETAILS_FALIURE";
export const fetchServiceDetailsFaliure = err => ({
  type: "FETCH_SERVICE_DETAILS_FALIURE",
  err
});

export const UPDATE_SERVICE_DETAILS = "UPDATE_SERVICE_DETAILS";
export const updateServiceDetails = data => ({
  type: UPDATE_SERVICE_DETAILS,
  data
});

export const UPDATE_SERVICE_DETAILS_SUCCESS = "UPDATE_SERVICE_DETAILS_SUCCESS";
export const updateServiceDetailsSuccess = service => ({
  type: UPDATE_SERVICE_DETAILS_SUCCESS,
  service
});

export const UPDATE_SERVICE_DETAILS_FALIURE = "UPDATE_SERVICE_DETAILS";
export const updateServiceDetailsFaliure = err => ({
  type: UPDATE_SERVICE_DETAILS_FALIURE,
  err
});

export const DELETE_SERVICE = "DELETE_SERVICE";
export const deleteService = id => ({
  type: DELETE_SERVICE,
  id
});

export const DELETE_SERVICE_SUCCESS = "DELETE_SERVICE_SUCCESS";
export const deleteServiceSuccess = res => ({
  type: DELETE_SERVICE_SUCCESS,
  res
});

export const DELETE_SERVICE_FALIURE = "DELETE_SERVICE_FALIURE";
export const deleteServiceFaliure = err => ({
  type: DELETE_SERVICE_FALIURE,
  err
});

export const REMOVE_SERVICE = "UPDATE_SERVICE_VALUE";
export const removeService = () => ({
  type: REMOVE_SERVICE
});

export const TOGGLE_SERVICE = "TOGGLE_SERVICE";
export const toggleService = service => ({
  type: TOGGLE_SERVICE,
  service
});

export const TOGGLE_SERVICE_SUCCESS = "TOGGLE_SERVICE_SUCCESS";
export const toggleServiceSuccess = () => ({
  type: TOGGLE_SERVICE_SUCCESS
});

export const TOGGLE_SERVICE_FALIURE = "TOGGLE_SERVICE_FALIURE";
export const toggleServiceFaliure = () => ({
  type: TOGGLE_SERVICE_FALIURE
});
