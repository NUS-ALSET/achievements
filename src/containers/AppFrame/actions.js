export const LOGIN_MENU_OPEN = "LOGIN_MENU_OPEN";
export const loginMenuOpen = anchorElId => ({
  type: LOGIN_MENU_OPEN,
  anchorElId
});

export const LOGIN_MENU_CLOSE = "LOGIN_MENU_CLOSE";
export const loginMenuClose = () => ({
  type: LOGIN_MENU_CLOSE
});

export const MAIN_DRAWER_TOGGLE = "MAIN_DRAWER_TOGGLE";
export const mainDrawerToggle = status => ({
  type: MAIN_DRAWER_TOGGLE,
  status
});

export const GET_DYNAMIC_PATHTITLE = "GET_DYNAMIC_PATHTITLE";
export const getDynamicPathtitle = pathname => ({
  type: GET_DYNAMIC_PATHTITLE,
  pathname
});

export const SAVE_PROMO_CODE = "SAVE_PROMO_CODE";
export const savePromoCode = code => ({
  type: SAVE_PROMO_CODE,
  code
});

export const ROUTES_CHANGED = "ROUTES_CHANGED";
export const routeChanged = pathName => ({
  type: ROUTES_CHANGED,
  pathName
});
