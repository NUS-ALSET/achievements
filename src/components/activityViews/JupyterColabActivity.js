/**
 * @file JupyterColabActivity container module
 * @author Theodor Shaytanov <theodor.shaytanov@gmail.com>
 * @created 08.03.18
 */

import React, { Fragment } from "react";
import PropTypes from "prop-types";

import { problemSolveUpdate } from "../../containers/Activity/actions";
import JupyterNotebook from "./JupyterNotebook";

class JupyterColabActivity extends React.PureComponent {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    onChange: PropTypes.func,
    problem: PropTypes.object.isRequired,
    solution: PropTypes.object,
    readOnly : PropTypes.bool
  };

  onSolutionRefreshClick = solutionURL => {
    const { dispatch, problem } = this.props;

    if (this.props.onChange) {
      this.props.onChange(solutionURL);
    }

    dispatch(
      problemSolveUpdate(problem.pathId, problem.problemId, solutionURL)
    );
  };

  render() {
    const {
      /** @type {JupyterPathProblem} */
      problem,
      solution,
      readOnly
    } = this.props;

    return (
      <Fragment>
        <JupyterNotebook
          action={this.onSolutionRefreshClick}
          defaultValue={solution && solution.id}
          persistent={true}
          solution={solution}
          readOnly={readOnly}
          title="Calculated Solution"
        />
        {solution &&
          solution.provided && (
            <JupyterNotebook
              solution={{
                json: solution.provided
              }}
              title="Provided Solution"
              readOnly={readOnly}
            />
          )}
        <JupyterNotebook
          solution={{ json: problem.problemJSON }}
          title="Problem"
          url={problem.problemColabURL}
          readOnly={readOnly}
        />
      </Fragment>
    );
  }
}

export default JupyterColabActivity;
