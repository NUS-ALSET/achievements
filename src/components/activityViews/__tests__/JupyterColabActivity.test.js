/* eslint-disable no-magic-numbers */
import React from "react";
import { createShallow } from "@material-ui/core/test-utils";

import sinon from "sinon";

import JupyterColabActivity from "../JupyterColabActivity";
import JupyterNotebook from "../JupyterNotebook";
import { PROBLEM_SOLVE_UPDATE } from "../../../containers/Activity/actions";

describe("<JupyterColabActivity>", () => {
  let mockDispatch;
  let shallow;

  beforeEach(() => {
    mockDispatch = sinon.spy();
    shallow = createShallow();
  });

  it("should check snapshot", () => {
    const wrapper = shallow(
      <JupyterColabActivity
        dispatch={mockDispatch}
        problem={{ pathId: "testPath", problemId: "testProblem" }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should initiate JupyterColabActivity w/o solution", () => {
    const wrapper = shallow(
      <JupyterColabActivity
        dispatch={mockDispatch}
        problem={{
          id: "testProblem",
          problemJSON: {
            foo: "bar"
          }
        }}
      />
    );

    expect(wrapper.find(JupyterNotebook).length).toEqual(2);
    wrapper.update();

    const solutionNotebook = wrapper.find(JupyterNotebook).at(0);
    const problemNotebook = wrapper.find(JupyterNotebook).at(1);

    expect(solutionNotebook.prop("persistent")).toEqual(true);
    expect(solutionNotebook.prop("title")).toEqual("Calculated Solution");
    expect(problemNotebook.props()).toEqual({
      solution: { json: { foo: "bar" } },
      title: "Problem"
    });
  });

  it("should initiate JupyterColabActivity with solution", () => {
    const wrapper = shallow(
      <JupyterColabActivity
        dispatch={mockDispatch}
        problem={{
          id: "testProblem",
          problemJSON: {
            foo: "bar"
          }
        }}
        solution={{
          json: "test",
          provided: "providedSolution"
        }}
      />
    );

    expect(wrapper.find(JupyterNotebook).length).toEqual(3);
    wrapper.update();

    const solutionNotebook = wrapper.find(JupyterNotebook).at(0);
    const providedNotebook = wrapper.find(JupyterNotebook).at(1);
    const problemNotebook = wrapper.find(JupyterNotebook).at(2);

    expect(solutionNotebook.prop("persistent")).toEqual(true);
    expect(solutionNotebook.prop("title")).toEqual("Calculated Solution");
    expect(solutionNotebook.prop("solution")).toEqual({
      json: "test",
      provided: "providedSolution"
    });

    expect(providedNotebook.prop("title")).toEqual("Provided Solution");
    expect(providedNotebook.prop("solution")).toEqual({
      json: "providedSolution"
    });

    expect(problemNotebook.props()).toEqual({
      solution: { json: { foo: "bar" } },
      title: "Problem"
    });
  });

  it("should dispatch problemSolutionSubmitRequest", () => {
    const mockChange = sinon.spy();
    const wrapper = shallow(
      <JupyterColabActivity
        dispatch={mockDispatch}
        onChange={mockChange}
        problem={{ pathId: "testPath", problemId: "testProblem" }}
      />
    );
    wrapper.instance().onSolutionRefreshClick("test");
    expect(mockChange.calledWith("test")).toEqual(true);
    expect(
      mockDispatch.calledWith({
        type: PROBLEM_SOLVE_UPDATE,
        pathId: "testPath",
        problemId: "testProblem",
        fileId: "test"
      })
    ).toEqual(true);
  });
});
