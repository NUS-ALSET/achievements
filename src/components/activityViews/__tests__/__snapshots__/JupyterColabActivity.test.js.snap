// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JupyterColabActivity> should check snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JupyterColabActivity
    dispatch={[Function]}
    problem={
        Object {
            "pathId": "testPath",
            "problemId": "testProblem",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <JupyterNotebook
          action={[Function]}
          defaultValue={undefined}
          persistent={true}
          readOnly={undefined}
          solution={undefined}
          title="Calculated Solution"
/>,
        undefined,
        <JupyterNotebook
          readOnly={undefined}
          solution={
                    Object {
                              "json": undefined,
                            }
          }
          title="Problem"
          url={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "action": [Function],
          "defaultValue": undefined,
          "persistent": true,
          "readOnly": undefined,
          "solution": undefined,
          "title": "Calculated Solution",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "readOnly": undefined,
          "solution": Object {
            "json": undefined,
          },
          "title": "Problem",
          "url": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <JupyterNotebook
            action={[Function]}
            defaultValue={undefined}
            persistent={true}
            readOnly={undefined}
            solution={undefined}
            title="Calculated Solution"
/>,
          undefined,
          <JupyterNotebook
            readOnly={undefined}
            solution={
                        Object {
                                    "json": undefined,
                                  }
            }
            title="Problem"
            url={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": [Function],
            "defaultValue": undefined,
            "persistent": true,
            "readOnly": undefined,
            "solution": undefined,
            "title": "Calculated Solution",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "readOnly": undefined,
            "solution": Object {
              "json": undefined,
            },
            "title": "Problem",
            "url": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
